shell.nix

```diff
@@ -1 +1 @@
-(import ./release.nix).grpc-haskell.env
+(import ./release.nix).grpc-haskell-core.env
```

Comment:
I think we want to keep this file the same, but add a `shell.nix` appropriate for `grpc-haskell-core` within its subdirectory.

Reply:
fully agree, I had just used this for debugging.

Reply:
Core already has a `shell.nix` setup with `(import ../release.nix).grpc-haskell-core.env`
This change can probably be removed.